applicationset/generators/merge.go

```diff
@@ -59,55 +72,140 @@ func (m *MergeGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.Appl
 		return nil, ErrLessThanTwoGeneratorsInMerge
 	}
 
+	var joinType, err = getJoinType(appSetGenerator.Merge.Mode)
+	if err != nil {
+		return nil, err
+	}
+
 	paramSetsFromGenerators, err := m.getParamSetsForAllGenerators(appSetGenerator.Merge.Generators, appSet)
 	if err != nil {
 		return nil, fmt.Errorf("error getting param sets from generators: %w", err)
 	}
 
-	baseParamSetsByMergeKey, err := getParamSetsByMergeKey(appSetGenerator.Merge.MergeKeys, paramSetsFromGenerators[0])
+	baseParamSetsByMergeKey := make(map[string][]map[string]interface{})
+	if strings.HasSuffix(string(joinType), UniqJoinSuffix) {
+		baseParamSetsByMergeKey, err = getParamSetsByMergeKey(appSetGenerator.Merge.MergeKeys, paramSetsFromGenerators[0], false)
+	} else {
+		baseParamSetsByMergeKey, err = getParamSetsByMergeKey(appSetGenerator.Merge.MergeKeys, paramSetsFromGenerators[0], true)
+	}
+
```

Comment:
If condition can be simplified with inline boolean variable

```suggestion
	isUniqueJoin := strings.HasSuffix(string(joinType), UniqJoinSuffix)
	baseParamSetsByMergeKey, err := getParamSetsByMergeKey(appSetGenerator.Merge.MergeKeys, paramSetsFromGenerators[0], !isUniqueJoin)
```